### PR TITLE
Bug/4402 cancel previous search requests

### PR DESCRIPTION
**Pull request by hussain-t**
## Summary

Addresses issue #4402 

## Checklist

- [x] My code is tested and passes existing unit tests.
- [x] My code has an appropriate set of unit tests which all pass.
- [x] My code is backward-compatible with WordPress 4.7 and PHP 5.6.
- [x] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [x] My code has proper inline documentation.
- [x] I have added a QA Brief on the issue linked above.
- [x] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

### CODE REVIEW

File: assets/js/components/PostSearcherAutoSuggest.js [hunk 1]
```diff
@@ -97,16 +97,27 @@ export default function PostSearcherAutoSuggest( {
...
 				{ query: encodeURIComponent( debouncedValue ) },
-				{ useCache: false }
+				{ useCache: false, signal: controller?.signal }
 			)
 				.then( ( res ) => setResults( res ) )
 				.catch( () => setResults( [] ) )
 				.finally( () => setIsLoading?.( false ) );
+
+			// Clean-up abort
+			return () => controller.abort();
```
Comment (eugene-manuilov) [inline assets/js/components/PostSearcherAutoSuggest.js:120]: This will throw an exception if `controller` is `undefined`.

**Review by eugene-manuilov — CHANGES_REQUESTED** [review]
Thanks, @hussain-t. This almost looks good. Added one comment and this should be good to go.

File: assets/js/components/PostSearcherAutoSuggest.js (hunk 1 above)
Comment (hussain-t) [reply assets/js/components/PostSearcherAutoSuggest.js:120]: Good catch. Apologies I missed the optional chaining `controller?.abort();`. Thanks @eugene-manuilov

**Review by eugene-manuilov — APPROVED** [review]
Thanks, @hussain-t. LGTM!

**Pull request merged**